### PR TITLE
chore(connector-wechat*): fix wechat connectors error handler, comments and UTs

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Fix WeChat connectors error handler, comments, and UTs.

## Linear Issue Reference

LOG-2753

## Testing

Passed UTs.

### CODE REVIEW

File: packages/connector-wechat-native/src/index.ts [hunk 1]
```diff
@@ -68,17 +65,16 @@ export default class WeChatNativeConnector implements SocialConnector {
...
         timeout: defaultTimeout,
       })
       .json<AccessTokenResponse>();
 
-    assert(
-      errcode !== 40_029 && accessToken && openid,
-      new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid)
-    );
+    assert(errcode !== 40_029, new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+    assert(!errcode && accessToken && openid, new Error(errmsg));
```
Comment (IceHe) [inline packages/connector-wechat-native/src/index.ts:77]: just curious: better use errcode when errmsg is blank?

should align with below `errmsg ?? ''`

File: packages/connector-wechat-native/src/index.test.ts [hunk 2]
```diff
@@ -57,15 +57,25 @@ describe('getAccessToken', () => {
     expect(openid).toEqual('openid');
   });
 
-  it('throws SocialAuthCodeInvalid error if accessToken not found in response', async () => {
+  it('throws SocialAuthCodeInvalid error if errcode is 40029', async () => {
     nock(accessTokenEndpointUrl.origin)
       .get(accessTokenEndpointUrl.pathname)
       .query(parameters)
-      .reply(200, {});
+      .reply(200, { errcode: 40_029, errmsg: 'invalid code' });
```
Comment (simeng-li) [inline packages/connector-wechat-native/src/index.test.ts:64]: just curious: `errorcode` and `errmsg`, are those WeChat reserved keys? neither snake case nor camel case. looks so weird.

File: packages/connector-wechat-native/src/index.ts [hunk 3]
```diff
@@ -107,6 +94,7 @@ export default class WeChatNativeConnector implements SocialConnector {
       // Response properties of user info can be separated into two groups: (1) {unionid, headimgurl, nickname}, (2) {errcode, errmsg}.
       // These two groups are mutually exclusive: if group (1) is not empty, group (2) should be empty and vice versa.
       // 'errmsg' and 'errcode' turn to non-empty values or empty values at the same time. Hence, if 'errmsg' is non-empty then 'errcode' should be non-empty.
+      // 'openid' is required but the response of getUserInfo is consistent as long as access_token is valid. In other words, 'openid' can be an arbitrary non-empty string, but it makes no sense that 'openid' can be some other strings other than its own value.
```
Comment (simeng-li) [inline packages/connector-wechat-native/src/index.ts:97]: not following this comment.  `openId` can be any string?  and Wechat identifies the requested userInfo by access_token only?

**Review by IceHe — APPROVED** [review]
overall lgtm.

File: packages/connector-wechat-native/src/index.ts (hunk 3 above)
Comment (darcyYe) [reply packages/connector-wechat-native/src/index.ts:97]: I've tried to call WeChat API with lots of different inputs to elaborate errors in different cases.
`openid` can be any non-empty string as long as `access_token` is valid, and it should not be undefined, null or empty string.

File: packages/connector-wechat-native/src/index.test.ts (hunk 2 above)
Comment (darcyYe) [reply packages/connector-wechat-native/src/index.test.ts:64]: They are WeChat API reserved keys for error messages.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**